src/rx-query.ts

```diff
@@ -490,6 +486,118 @@ export class RxQueryBase<
         this._limitBufferSize = bufferSize;
         return this;
     }
+
+    enablePersistentQueryCache(backend: QueryCacheBackend) {
+        this._queryCacheBackend = backend;
+        this._persistentQueryCacheLoaded = this._restoreQueryCacheFromPersistedState();
+        return this;
+    }
+
+    private async _restoreQueryCacheFromPersistedState() {
+        if (!this._queryCacheBackend) {
+            // no cache backend provided, do nothing
+            return;
+        }
+
+        if (this._persistentQueryCacheResult) {
+            // we already restored the cache once, no need to run twice
+            return;
+        }
+
+        if (this.mangoQuery.skip) {
+            console.error('The persistent query cache only works on non-skip queries.');
+            return;
+        }
+
+        const persistentQueryId = this.persistentQueryId();
+        const value = await this._queryCacheBackend.getItem<string[] | string>(`qc:${persistentQueryId}`);
+
+        const lwt = (await this._queryCacheBackend.getItem(`qc:${persistentQueryId}:lwt`)) as string ?? RX_META_LWT_MINIMUM;
+
+        const primaryPath = this.collection.schema.primaryPath;
+
+        this._persistentQueryCacheResult = value ?? undefined;
+
+        // if this is a regular query, also load documents into cache
+        if (Array.isArray(value) && value.length > 0) {
+            this._persistentQueryCacheResult = value;
+            this._persistentQueryCacheResultLwt = lwt;
+
+            const persistedQueryCacheIds = new Set(this._persistentQueryCacheResult);
+
+            let docsData: RxDocumentData<RxDocType>[] = [];
+
+            // query all docs updated > last persisted, limit to an arbitrary 1_000_000 (10x of what we consider our largest library)
+            const {documents: changedDocs} = await this.collection.storageInstance.getChangedDocumentsSince(
```

Comment:
is this a pretty fast and well indexed query? i'm assuming it is since you're gonna be running it for each query using pers. cache

Reply:
`[lwt+id]` is a built-in default index on every storage.

Reply:
This is doing the same, I am good with either.

```ts
const nonRestoredDocIds: string[] = [];
for (const docId of persistedQueryCacheIds) {
  // first try to fill from docCache
  const docData = this.collection._docCache.getLatestDocumentDataIfExists(docId);
  if (docData && this.doesDocumentDataMatch(docData)) {
    docsData.push(docData);
  }

  if (!docData) {
    nonRestoredDocIds.push(docId);
  }
}
```

Reply:
excellent